### PR TITLE
Feature/SOF-6690: lint & reformat

### PR DESCRIPTION
**Pull request by mrakitin**
Uses https://github.com/Exabyte-io/api-client/blob/dev/.github/workflows/cicd.yml as an example.

~~Based on PR #73 branch commits (needs a rebase on `dev` once #72 and #73 are merged)~~ ✔️ .

### CODE REVIEW

**Comment by mrakitin** [comment]
`ruff` configuration is based on https://beta.ruff.rs/docs/configuration/
`black` - https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#configuration-via-a-file

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      - id: black
+      - id: isort
+      - id: mypy
+        exclude: ^examples\/config\.py
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
```
Comment (chu-k) [inline .pre-commit-config.yaml:16]: I'm personally OK with having hooks such as these basic ones (no config options) separate from our repo as is done here. Thoughts @timurbazhirov ?

**Review by chu-k — APPROVED** [review]
LGTM, pending @timurbazhirov

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mrakitin) [reply .pre-commit-config.yaml:16]: Thanks! Yes, those looked pretty simple, and the corresponding .py code lives in the same repo, so it's a bit too much work to port it to this org.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+exclude: ^((examples|other)\/.*\/.*\.(py|poscar|csv))
+repos:
+  - repo: https://github.com/mrakitin/exabyte-io-pre-commit-hooks
```
Comment (timurbazhirov) [inline .pre-commit-config.yaml:3]: Should be https://github.com/Exabyte-io/pre-commit-hooks, right? Based on https://github.com/Exabyte-io/pre-commit-hooks/pull/1/files

File: .pre-commit-config.yaml (hunk 1 above)
Comment (timurbazhirov) [reply .pre-commit-config.yaml:16]: I think it's better to build a unified setup for ourselves in one place and reuse it for all the repos from there. If we need to, we can include lines 11-16 in https://github.com/Exabyte-io/pre-commit-hooks

Comment (mrakitin) [reply .pre-commit-config.yaml]: https://github.com/Exabyte-io/pre-commit-hooks/pull/2 is to address the request.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (mrakitin) [reply .pre-commit-config.yaml:3]: Yes, eventually it should point to the https://github.com/Exabyte-io/pre-commit-hooks repo, preferably with a release as a `rev`.

Need to merge https://github.com/Exabyte-io/pre-commit-hooks/pull/2 to fully transition to that config. Right now it's still using my fork + commit hash.

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**